specification/redis/Cache.Management/RedisResource.tsp

```diff
@@ -0,0 +1,228 @@
+      @segment("redis")
+      @path
+      name: string;
+    },
+    RedisCreateParameters,
+    RedisResource | ArmResourceCreatedResponse<
+      RedisResource,
+      LroHeaders = ArmLroLocationHeader<FinalResult = RedisResource>
+    >,
+    ErrorResponse
+  >;
+
+  /**
+   * Update an existing Redis cache.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/no-response-body" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @patch(#{ implicitOptionality: false })
+  update is RedisResourcesOps.CustomPatchAsync<
+    RedisResource,
+    PatchModel = RedisUpdateParameters,
+    Response = ArmResponse<RedisResource> | (ArmAcceptedLroResponse & {
+      @bodyRoot
+      _: RedisResource;
+    })
+  >;
+
+  /**
+   * Deletes a Redis cache.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-delete-operation-response-codes" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  delete is RedisResourcesOps.DeleteWithoutOkAsync<
+    RedisResource,
+    Response = ArmDeletedResponse | ArmDeleteAcceptedLroResponse | ArmDeletedNoContentResponse
+  >;
+
+  /**
+   * Lists all Redis caches in a resource group.
+   */
+  listByResourceGroup is ArmResourceListByParent<
+    RedisResource,
+    Response = ArmResponse<RedisListResult>
+  >;
+
+  /**
+   * Gets all Redis caches in the specified subscription.
+   */
+  listBySubscription is ArmListBySubscription<
+    RedisResource,
+    Response = ArmResponse<RedisListResult>
+  >;
+
+  /**
+   * Gets any upgrade notifications for a Redis cache.
+   */
+  @get
+  listUpgradeNotifications is RedisResourcesOps.ActionSync<
+    RedisResource,
+    void,
+    ArmResponse<NotificationListResponse>,
+    Parameters = {
+      /**
+       * how many minutes in past to look for upgrade notifications
+       */
+      @query("history")
+      history: float64;
+    }
+  >;
+
+  /**
+   * Retrieve a Redis cache's access keys. This operation requires write permission to the cache resource.
+   */
+  listKeys is RedisResourcesOps.ActionSync<
+    RedisResource,
+    void,
+    ArmResponse<RedisAccessKeys>
+  >;
+
+  /**
+   * Regenerate Redis cache's access keys. This operation requires write permission to the cache resource.
+   */
+  regenerateKey is RedisResourcesOps.ActionSync<
+    RedisResource,
+    RedisRegenerateKeyParameters,
+    ArmResponse<RedisAccessKeys>
+  >;
+
+  /**
+   * Reboot specified Redis node(s). This operation requires write permission to the cache resource. There can be potential data loss.
+   */
+  forceReboot is RedisResourcesOps.ActionSync<
+    RedisResource,
+    RedisRebootParameters,
+    ArmResponse<RedisForceRebootResponse>
+  >;
+
+  /**
+   * Import data into Redis cache.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-post-operation-response-codes" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  @action("import")
```

Comment:
Justification would be backCompat right?
What is this actually complaining about though??